actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -358,6 +358,7 @@ def body_stream # :nodoc:
 
     def reset_session
       session.destroy
+      controller_instance.reset_csrf_token(self) if controller_instance.respond_to?(:reset_csrf_token)
```

Comment:
Should this happen _before_ the session is destroyed? I'm not sure it will have any real difference, probably not, but logically speaking if the CSRF is in the session I'd think we should perform that delete action first, rather than after destroying the session.

Reply:
I feel like I went thorough this thought process and did it this way on purpose, but honestly can't remember why.  There may be some sort of side effect in `session.destroy` that would have interacted in a weird way with the `CookieStore`.

Reply:
Changing the order doesn't seem to break any action pack tests locally, but it's possible some other side effect you might've come across would show up -- I can't think of any off the top of my head.

I pushed the method extraction here: https://github.com/rails/rails/commit/b925880914fa3dca8c9cd0f8e88fb18fc8ec180b.

Reply:
Took a quick peek.  Doesn't appear that there are any side effects.  There were a lot of iterations on this PR, so I'm likely remembering something from an earlier attempt.